tests/kafkatest/services/kafka/kafka.py

```diff
@@ -778,7 +781,14 @@ def prop_file(self, node):
 
         if self.use_share_groups is not None and self.use_share_groups is True:
             override_configs[config_property.SHARE_GROUP_ENABLE] = str(self.use_share_groups)
-            override_configs[config_property.GROUP_COORDINATOR_REBALANCE_PROTOCOLS] = 'classic,consumer,share'
+            if self.use_streams_groups is True:
+                override_configs[config_property.UNSTABLE_API_VERSIONS_ENABLE] = str(True)
+                override_configs[config_property.GROUP_COORDINATOR_REBALANCE_PROTOCOLS] = 'classic,consumer,share,streams'
+            else:
+                override_configs[config_property.GROUP_COORDINATOR_REBALANCE_PROTOCOLS] = 'classic,consumer,share'
+        elif self.use_streams_groups is True:
+            override_configs[config_property.UNSTABLE_API_VERSIONS_ENABLE] = str(True)
+            override_configs[config_property.GROUP_COORDINATOR_REBALANCE_PROTOCOLS] = 'classic,consumer,streams'
```

Comment:
Would it be simpler to start with `enabledProtocols = 'classic,consumer'` and do
```
if self.use_streams_groups is True:
  enabledProtocols += ',streams`
```
and end with
```
override_configs[config_property.GROUP_COORDINATOR_REBALANCE_PROTOCOLS] = enabledProtocols
```
Similar for share group? To avoid all this complicated nesting?